### PR TITLE
mergeTableData for New Storage Format

### PR DESCRIPTION
**Pull request by druvv**
Implements the `mergeTableData` function for the new storage format. Any conflicts will throw an error. Cell-wise merge is implemented.

TODO:
- Fix merging of secondary indexes in the presence of cell-wise merges. 
- Conflicts

### CODE REVIEW

File: go/libraries/doltcore/merge/merge.go [hunk 1]
```diff
@@ -136,9 +140,12 @@ func (merger *Merger) MergeTable(ctx context.Context, tblName string, opts edito
 		if rootHash == ancHash {
 			ms := MergeStats{Operation: TableModified}
 			if rootHash != mergeHash {
-				ms, err = calcTableMergeStats(ctx, tbl, mergeTbl)
-				if err != nil {
-					return nil, nil, err
+				// TODO (dhruv): calculate stats for V1
+				if merger.vrw.Format() != types.Format_DOLT_1 {
```
Comment (andrew-wm-arthur) [inline go/libraries/doltcore/merge/merge.go:144]: move this format check down into `calcTableMergeStats`

File: go/libraries/doltcore/merge/merge.go [hunk 2]
```diff
@@ -230,6 +259,128 @@ func (merger *Merger) MergeTable(ctx context.Context, tblName string, opts edito
...
+		if err != nil {
+			return nil, err
+		}
+		ancI, ancOK, err := anc.GetIndex(ctx, postMergeSchema, index.Name())
+		if err != nil {
+			return nil, err
+		}
+
+		mergedIndex, err := func() (durable.Index, error) {
+			// Build a new index based on |mergedData| if an index is missing.
```
Comment (andrew-wm-arthur) [inline go/libraries/doltcore/merge/merge.go:348]: there is already logic to do this in `MergeTable`, do we need to duplicate it here?

**Review by andrew-wm-arthur — COMMENTED** [review]
general approach looks good, needs a few tweaks and to fix the failing tests

File: go/libraries/doltcore/merge/merge.go (hunk 2 above)
Comment (druvv) [reply go/libraries/doltcore/merge/merge.go:348]: yes. In the new format, we are building this added index based on `mergedData`, which is the post-merge primary row data.

In the old format, we are building this added index based on the pre-merge data. Then we use the tblEditor's functionality to bring it up to date with the merged data.

File: go/store/prolly/map.go [hunk 3]
```diff
@@ -270,3 +273,14 @@ func DebugFormat(ctx context.Context, m Map) (string, error) {
...
+func MustDebugFormat(t *testing.T, m Map) string {
+	s, err := DebugFormat(context.Background(), m)
+	require.NoError(t, err)
+	return s
+}
+
+func MustEqual(t *testing.T, expected Map, actual Map) {
+	require.Equal(t, expected.HashOf(), actual.HashOf(),
+		"hashes differed. expected: %s\nactual: %s", MustDebugFormat(t, expected), MustDebugFormat(t, actual))
+}
```
Comment (andrew-wm-arthur) [inline go/store/prolly/map.go:286]: these don't belong here

**Review by andrew-wm-arthur — APPROVED** [review]
LGTM!

**Pull request merged**